### PR TITLE
split_file_to_collection: add txt

### PR DESCRIPTION
**Pull request by bernt-matthias**
fixes https://github.com/bgruening/galaxytools/issues/886

### CODE REVIEW

**Comment by bgruening** [comment]
> Failed to find identifier [test_1.txt] for testing, tool generated collection elements [[]]

**Comment by bernt-matthias** [comment]
On my local planemo tests for some reason my local python 2.7 is executed which does not know about `math.inf`

**Comment by bgruening** [comment]
```
--- local_file
+++ history_data
@@ -1,4 +1,3 @@
-#This is a file
 #file   data
 foo2.mgf	bar2
 foo.mgf	bar4
```

Mh, math should be part of the standard lib :(

**Comment by bernt-matthias** [comment]
Hey @bgruening what do you think about the generic splitter. 

I would have liked to set the format identical to the input, but currently only dynamic detection works: https://github.com/galaxyproject/galaxy/issues/2431

Also a macro for the regexp validator. If you like we may also remove the txt case (since generic defaults to this behaviour). 

Also was thinking that I would like to remove the extension from the element identifiers.

File: tools/text_processing/split_file_to_collection/split_file_to_collection.xml [hunk 1]
```diff
@@ -1,6 +1,19 @@
-<tool id="split_file_to_collection" name="Split file" version="0.1.1">
+<tool id="split_file_to_collection" name="Split file" version="0.2.0">
     <description>to dataset collection</description>
     <macros>
+        <xml name="regex_sanitizer">
```
Comment (bgruening) [inline tools/text_processing/split_file_to_collection/split_file_to_collection.xml]: I like that!

File: tools/text_processing/split_file_to_collection/split_file_to_collection.xml [hunk 2]
```diff
@@ -73,6 +91,8 @@
                 <option value="fastq">FASTQ</option>
                 <option value="tabular">Tabular</option>
                 <option value="fasta">FASTA</option>
+                <option value="txt">Text files</option>
+                <option value="generic">Genric</option>
```
Comment (bgruening) [inline tools/text_processing/split_file_to_collection/split_file_to_collection.xml]: This should be generic I guess ... maybe a little bit more explanation ...?
Also you might want to add an example to the help?

Comment (bernt-matthias) [reply tools/text_processing/split_file_to_collection/split_file_to_collection.xml]: Sure. I extended the help text. Is this sufficient in your opinion?

**Comment by bgruening** [comment]
Thanks @bernt-matthias!

**Pull request merged**